test/e2e/docker-compose.yml

```diff
@@ -0,0 +1,84 @@
+version: "3"
+services:
+  zookeeper:
+    image: confluentinc/cp-zookeeper@sha256:87314e87320abf190f0407bf1689f4827661fbb4d671a41cba62673b45b66bfa
+    ports:
+      - "2181:2181"
+    environment:
+      ZOOKEEPER_CLIENT_PORT: 2181
+      ZOOKEEPER_TICK_TIME: 2000
+      ZOOKEEPER_SYNC_LIMIT: 2
+
+  kafka:
+    image: confluentinc/cp-kafka@sha256:c6320f9a0cbf57075e102546de110dcebdf374955f12388d58c23a54b8a47d31
+    ports:
+      - 9094:9094
+    depends_on:
+      - zookeeper
+    environment:
+      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
+      KAFKA_LISTENERS: INTERNAL://kafka:9092,OUTSIDE://kafka:9094
+      KAFKA_ADVERTISED_LISTENERS: INTERNAL://kafka:9092,OUTSIDE://localhost:9094
+      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: INTERNAL:PLAINTEXT,OUTSIDE:PLAINTEXT
+      KAFKA_INTER_BROKER_LISTENER_NAME: INTERNAL
+      KAFKA_offsets_topic_replication_factor: 1
+
+  minio:
+    image: minio/minio@sha256:684ce208c005fe032659ec77bafa6a17a16c41686c334618dec924b3505e7090
+    hostname: minio
+    ports:
+      - 9000:9000
+    environment:
+      MINIO_ROOT_USER: minio
+      MINIO_ROOT_PASSWORD: minio123
+      MINIO_REGION_NAME: dummy_region
+    entrypoint: sh
+    command: -c 'mkdir -p /data/package-analysis{,-static,-file-writes} && /usr/bin/minio server /data'
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
+      interval: 30s
+      timeout: 20s
+      retries: 3
+
+  analysis:
+    image: gcr.io/ossf-malware-analysis/analysis:test
+    privileged: true
+    entrypoint: "/usr/local/bin/worker"
+    environment:
+      OSSMALWARE_WORKER_SUBSCRIPTION: kafka://worker?topic=workers
+      OSSF_MALWARE_NOTIFICATION_TOPIC: kafka://notifications
+      OSSF_MALWARE_ANALYSIS_RESULTS: s3://package-analysis?endpoint=minio:9000&disableSSL=true&s3ForcePathStyle=true
+      OSSF_MALWARE_STATIC_ANALYSIS_RESULTS: s3://package-analysis-static?endpoint=minio:9000&disableSSL=true&s3ForcePathStyle=true
+      OSSF_MALWARE_ANALYSIS_FILE_WRITE_RESULTS: s3://package-analysis-file-writes?endpoint=minio:9000&disableSSL=true&s3ForcePathStyle=true
+      OSSF_SANDBOX_NOPULL: "true"
```

Comment:
Is this file essentially the same as https://github.com/ossf/package-analysis/blob/main/examples/e2e/docker-compose.yml  with the exception of this? 

Can we instead use something like https://docs.docker.com/compose/environment-variables/set-environment-variables/#substitute-with---env-file to avoid too much duplication?

Reply:
Yeah the file overall is pretty much the same. Other than the NOPULL options, the image tags are also different (`:test`). It would be nice to reduce the duplication but I'm not sure how much more can be deduplicated other than environment vars. Also, the examples/e2e folder seems like a bit of a free-for-all so maybe it's okay to have two copies.

Reply:
Looks like there's a way to override anything (e.g. image tags) via e.g. https://docs.docker.com/compose/extends/#example-use-case as well. 

Reducing duplication seems better to me -- let's avoid having two diverging but very similar configs around if we can.

Reply:
Nice find! Let's do that then. I might make a new directory to hold the base config because it doesn't seem right to put it either in `test/` or `examples/`. Perhaps `deploy/docker-compose`?

Reply:
Leaving it in `examples/` seems OK to me. Alternatively, maybe `/configs`: https://github.com/golang-standards/project-layout#configs. Up to you, I don't have a strong opinion here on the colour of this bikeshed :)